python/mxnet/ndarray/ndarray.py

```diff
@@ -903,29 +992,24 @@ def _get_nd_basic_indexing(self, key):
                     'index {} is out of bounds for axis {} with size {}'
                     ''.format(key_nd[ax], ax, self.shape[ax]))
 
-        # Make sure we don't accidentally have advanced indexing or
-        # unsupported entries.
-        for idx in slc_key:
-            if not isinstance(idx, py_slice):
-                raise RuntimeError(
-                    'found object of type {} instead of `slice`. '
-                    'This is a bug, please report it!'
-                    ''.format(type(idx)))
-
         # Convert to begin, end and step, and return immediately if the slice
         # is empty
         begin, end, step = self._basic_indexing_key_to_begin_end_step(
             slc_key, self.shape, keep_none=False
         )
-        # Pylint is wrong about this
-        # pylint: disable=bad-continuation
-        if any(
-            b >= e and s > 0 or b <= e and s < 0 for b, e, s in zip(begin, end, step)
-        ):
-            return array([], self.context, self.dtype)
-        # pylint: enable=bad-continuation
-
-        if self._basic_indexing_slice_is_contiguous(slc_key, self.shape):
+
+        # handle zero-size outputs
```

Comment:
There is no need to support zero-size output in old `NDArray`. For such cases, the `mx.nd.slice` should terminate with infer-shape error already.

Reply:
Will fix. Thx!